rvic/convolution.py

```diff
@@ -49,20 +52,22 @@ def convolution(config):
     try:
         # ---------------------------------------------------------------- #
         # Initilize
-        hist_tapes, data_model, rout_var, time_handle, directories = convolution_init(
+        hist_tapes, data_model, rout_var, time_handle, directories, shared_timestamp, listener_thread, event = convolution_init(
             config
         )
         # ---------------------------------------------------------------- #
 
         # ---------------------------------------------------------------- #
         # Run
         time_handle, hist_tapes = convolution_run(
-            hist_tapes, data_model, rout_var, time_handle, directories
+            hist_tapes, data_model, rout_var, time_handle, directories, shared_timestamp
         )
         # ---------------------------------------------------------------- #
 
         # ---------------------------------------------------------------- #
         # Finalize
+        event.set()
+        listener_thread.join()
```

Comment:
Should these be executed in the `finally` block? I.e. I think that we want them to run *always*, and as written here, they won't if some exception is raised by `convolution_run`. Speaking of which... why is there no `except` block corresponding to the `try`. Will this code will silently fail?